### ISSUE TITLE
Feature Request: Firefox Android Support

### ISSUE THREAD

**Issue by Starfiresg1** [wanhose/cookie-dialog-monster#828]
Hello,

since Mozilla relaxed the Addon-Limits on Firefox for Android and it's possible to install Addons from XPI-Files now (with a hidden menu) I gave Cookie Dialog Monster a try. While the addon installed and shows up in the the Addon-Menu it sadly it doesn't seem to work at all at the moment.

Maybe this is something that can be looked into at some point?

**Comment by wanhose** [comment]
Hi @Starfiresg1,

I just marked the option `compatible for Firefox Android` while submitting the new `7.1.5` version to the Mozilla team. I don't know if it has something to do with the issue you are describing, there are users already using this extension on `Kiwi Browser Android` for example.

Please, let me know if the new version solves the issue for you. If not, can you attach some screenshots/steps in how to reproduce the issue?

Thanks for you valuable feedback! 🥇

**Comment by Starfiresg1** [comment]
Hello,
I updated the addon with the current "latest.xpi" from Downloads.
![Screenshot_2024-08-26-21-47-52-551_org mozilla firefox](https://github.com/user-attachments/assets/1e88e8fe-724e-4865-ba8f-939ff0ef2e30)

The addon is listed in the addon menu
![Screenshot_2024-08-26-21-47-16-550_org mozilla firefox](https://github.com/user-attachments/assets/ee0cbd2b-f370-468b-9b3e-be06b0589e36)

Nothing happens when I click on it - so I can't access the site specific settings (disable for specific site, reporting errors etc.) or update the database.
Futhermore I don't noticed any cookie dialogs being blocked - even on popular sites like Google.

Firefox Android is the current version - addon was installed from file through the hidden developer menu.
![Screenshot_2024-08-26-21-54-26-471_org mozilla firefox](https://github.com/user-attachments/assets/5e38e9c5-9350-4c2c-9e92-68b2a4b7df42)

**Comment by wanhose** [comment]
> Hello, I updated the addon with the current "latest.xpi" from Downloads. <img alt="" width="50%" height="50%" src="https://private-user-images.githubusercontent.com/8659009/361556861-1e88e8fe-724e-4865-ba8f-939ff0ef2e30.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwOTUyOTMsIm5iZiI6MTcyNTA5NDk5MywicGF0aCI6Ii84NjU5MDA5LzM2MTU1Njg2MS0xZTg4ZThmZS03MjRlLTQ4NjUtYmE4Zi05MzlmZjBlZjJlMzAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzFUMDkwMzEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTA0MjQ3NTI0ZmM1OGNiOWJmZWE2OTBkNmRiNTljMzZmMmJhMmMyZGE5Y2IwZjQ4ZjljMDNjMjI5MGUzNDhiYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Ty0-dkXqM7FOqXH8S0lzoUOurFOCGa2HWrSuOZttmes">
> 
> The addon is listed in the addon menu <img alt="" width="50%" height="50%" src="https://private-user-images.githubusercontent.com/8659009/361556972-ee0cbd2b-f370-468b-9b3e-be06b0589e36.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwOTUyOTMsIm5iZiI6MTcyNTA5NDk5MywicGF0aCI6Ii84NjU5MDA5LzM2MTU1Njk3Mi1lZTBjYmQyYi1mMzcwLTQ2OGItOWIzZS1iZTA2YjA1ODllMzYuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzFUMDkwMzEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWVhOTczZGIwMjY0Y2JmNmIzYjhlN2M0MDJhNGJiZjdmZjllY2MyMDE0MGJhZjFlOTljOTA1YzJhNzdkYjMyMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NuYsGQa9AijrbeXNND_uhulD8qHSpjFl6QTyGLeZ0xM">
> 
> Nothing happens when I click on it - so I can't access the site specific settings (disable for specific site, reporting errors etc.) or update the database. Futhermore I don't noticed any cookie dialogs being blocked - even on popular sites like Google.
> 
> Firefox Android is the current version - addon was installed from file through the hidden developer menu. <img alt="" width="50%" height="50%" src="https://private-user-images.githubusercontent.com/8659009/361558415-5e38e9c5-9350-4c2c-9e92-68b2a4b7df42.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwOTUyOTMsIm5iZiI6MTcyNTA5NDk5MywicGF0aCI6Ii84NjU5MDA5LzM2MTU1ODQxNS01ZTM4ZTljNS05MzUwLTRjMmMtOWU5Mi02OGIyYTRiN2RmNDIuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzFUMDkwMzEzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDNhYzg5ODYzZTNiMDE0OGFhNTgwODMwNjc3ZDllYzY4ZGM5ZTEyMjg2MTM5MjM1YWJhYWY4ZWQ4MGQxNDZkZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RVX-v7LlwoyxCjd0i_LxMoupP8DJR2Mt3q0Ed7P7zcY">

Hi again @Starfiresg1,

I'm working on it, will keep you posted here! 👍

**Comment by wanhose** [comment]
Hi again @Starfiresg1,

I've been working on this from yesterday. I ended up making it work on Firefox for Android but there are some limitations due the poor work done by the Firefox team until today to make desktop extensions work on mobile devices. For example, Mozilla Mobile users won't be able to report broken sites.

Anyway, I've created a new branch that will host this new extension because it's basically a fork of the current extension with many modifications. Let's try to keep this manageable, but if this goes crazy at some point... we will drop support for Mozilla Mobile.

- [Installation guide for Mozilla Mobile users](https://github.com/wanhose/cookie-dialog-monster/tree/main/packages/browser-extension#installation-for-mozilla-firefox-mobile-users)
- [Latest Mozilla Mobile release](https://www.cookie-dialog-monster.com/releases/latest-mozilla-mobile.xpi) for you to easily download it
- [Mozilla Mobile main branch](https://github.com/wanhose/cookie-dialog-monster/tree/mozilla-mobile)

Thanks for this feature request again, I'm sure it will help many users to surf the Internet in a better way also on their mobile devices! 🚀
